cmake/helpers/CheckDependentLibraries.cmake

```diff
@@ -136,35 +165,19 @@ gdal_check_package(LibXml2 "Read and write XML formats" CAN_DISABLE)
 
 gdal_check_package(EXPAT "Read and write XML formats" RECOMMENDED CAN_DISABLE)
 gdal_check_package(XercesC "Read and write XML formats (needed for GMLAS and ILI drivers)" CAN_DISABLE)
-if (HAVE_EXPAT OR GDAL_USE_XERCESC)
-  set(HAVE_XMLPARSER ON)
-else ()
-  set(HAVE_XMLPARSER OFF)
-endif ()
 
 gdal_check_package(ZLIB "zlib (external)" CAN_DISABLE)
-if (NOT GDAL_USE_ZLIB)
-  set(GDAL_USE_ZLIB_INTERNAL
-      ON
-      CACHE BOOL "Use internal zlib copy (if set to ON, has precedence over GDAL_USE_ZLIB)")
-  if (NOT GDAL_USE_ZLIB_INTERNAL)
-    message(FATAL_ERROR "GDAL_USE_ZLIB or GDAL_USE_ZLIB_INTERNAL must be set to ON")
-  endif ()
-else ()
-  set(GDAL_USE_ZLIB_INTERNAL
-      OFF
-      CACHE BOOL "Use internal zlib copy (if set to ON, has precedence over GDAL_USE_ZLIB)")
+invert_on_off(GDAL_USE_ZLIB NOT_GDAL_USE_ZLIB)
+set(GDAL_USE_ZLIB_INTERNAL
+    ${NOT_GDAL_USE_ZLIB}
+    CACHE BOOL "Use internal zlib copy (if set to ON, has precedence over GDAL_USE_ZLIB)")
```

Comment:
I would generally prefer `option` for all the boolean variables: 
~~~
option(GDAL_USE_ZLIB_INTERNAL "${NOT_GDAL_USE_ZLIB}"
  "Use internal zlib copy (if set to ON, has precedence over GDAL_USE_ZLIB)")
~~~

Reply:
done